services/Donate.js

```diff
@@ -0,0 +1,112 @@
+const Donation = require("../models/Donation");
+
+class Donate {
+  /**
+   * @description create a new Flutterwave transaction
+   * @param {object} data
+   */
+  static async makeDonation(details) {
+    const url = "https://api.flutterwave.com/v3/payments";
+    const headers = {
+      Authorization: `Bearer ${process.env.FLW_SECRET_KEY}`,
+      "Content-Type": "application/json"
+    };
+    const fetchOptions = {
+      method: "POST",
+      headers,
+      body: JSON.stringify(details)
+    };
+
+    try {
+      const response = await fetch(url, fetchOptions);
+      const data = await response.json();
+
+      if (data.status === "error") {
+        return {
+          success: false,
+          error: data.errors
```

Comment:
We should log this errors on the backend for better debugging.

Reply:
Adding a todo for this.